posthog/warehouse/models/datawarehouse_saved_query.py

```diff
@@ -0,0 +1,75 @@
+from posthog.models.utils import UUIDModel, CreatedMetaFields, DeletedMetaFields
+from django.db import models
+from posthog.models.team import Team
+
+from posthog.hogql.database.models import SavedQuery
+from posthog.hogql.database.database import Database
+from typing import Dict
+import re
+
+from django.core.exceptions import ValidationError
+
+
+def validate_database_name(value):
+    if value in Database._table_names:
+        raise ValidationError(
+            f"{value} is not a valid view name. View names cannot overlap with PostHog table names.",
+            params={"value": value},
+        )
+
+
+class DatawarehouseSavedQuery(CreatedMetaFields, UUIDModel, DeletedMetaFields):
+    name: models.CharField = models.CharField(max_length=128, unique=True, validators=[validate_database_name])
```

Comment:
This cannot have `unique=True`, that means the name must be unique _globally_, across all projects in the PostHog instance. You need a `UniqueConstraint` instead.

Reply:
ah thanks for the catch

Reply:
Thanks for the catch!